src/nvim/ops.c

```diff
@@ -4654,13 +4697,34 @@ char_u get_reg_type(int regname, long *reglen)
   yankreg_T *reg = get_yank_register(regname, YREG_PASTE);
 
   if (reg->y_array != NULL) {
-    if (reglen != NULL && reg->y_type == MBLOCK)
-      *reglen = reg->y_width;
+    if (reg_width != NULL && reg->y_type == MBLOCK)
+      *reg_width = reg->y_width;
     return reg->y_type;
   }
   return MAUTO;
 }
 
+void format_reg_type(char_u reg_type, colnr_T reg_width, char* buf, size_t buf_len)
+{
+  assert(buf_len > 1);
+  buf[1] = NUL;
+  switch (reg_type) {
+    case MLINE:
+      buf[0] = 'V';
+      break;
+    case MCHAR:
+      buf[0] = 'v';
+      break;
+    case MBLOCK:
+      buf[0] = Ctrl_V;
+      snprintf(buf + 1, buf_len - 1, "%" PRIdCOLNR, reg_width + 1);
```

Comment:
What’s the point of setting `buf[0]` here and then messing with lengths? There is CTRL_V_STR macros so one can write `snprintf(buf, buf_len, CTRL_V_STR "%" PRIdCOLNR, reg_width + 1);`.